olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -363,13 +356,54 @@ bool DefaultCacheImpl::PromoteKeyLru(const std::string& key) {
   return true;
 }
 
+bool DefaultCacheImpl::PutMutableCache(const std::string& key,
+                                       const leveldb::Slice& value,
+                                       time_t expiry) {
+  if (!mutable_cache_) {
+    return true;
+  }
+
+  // can't put new item if cache is full and eviction disabled
+  const auto item_size = value.size();
+  if (!mutable_cache_lru_ &&
+      mutable_cache_data_size_ + item_size > settings_.max_disk_storage) {
+    return false;
+  }
+
+  if (IsExpiryValid(expiry)) {
+    auto time_size = StoreExpiry(key, *mutable_cache_, expiry);
+    if (time_size == 0) {
+      return false;
+    }
+    mutable_cache_data_size_ += time_size;
+  }
+  mutable_cache_data_size_ += key.size() + item_size;
```

Comment:
Maybe only count this if below Put is not false.
Also i would move expiry storing after the below Put, else you might land in a situation where you have an expiry but no data.